app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -86,8 +86,23 @@ EOF
 elif [ -f /etc/debian_version ]; then
   <%= save_to_file('/etc/apt/sources.list.d/foreman_registration.list', @repo) %>
 <% if @repo_gpg_key_url.present? -%>
+<%
+  <% # Ubuntu 16.04 does not support GPG public keys in "/etc/apt/trusted.gpg.d/" -%>
+  use_apt_key =
+    if @host.operatingsystem.name == 'Ubuntu' && @host.operatingsystem.major.split('.')[0].to_i == 16
+      true
+    else
+      false
+    end
```

Comment:
I'd simplify the logic and also make it `< 18`. That way you also catch older platforms, even if they're unsupported.
```suggestion
  use_apt_key = @host.operatingsystem.name == 'Ubuntu' && @host.operatingsystem.major.split('.')[0].to_i < 18
```

One edge case can be that `'a'.to_i` also equals 0. This can be mitigated by flipping it to `use_trusted_gpg = ... >= 18`, but that's likely to cause more problems than it solves.